### DISCUSSION TITLE
Conditional preview configuration based on current line

### DISCUSSION THREAD

**Discussion by danrlavoie** [junegunn/fzf#3884] [Q&A] [answered]
Hi, I've been trying to put together a file browser using fzf. I want to run search commands against both filepaths and against the file contents. For example, if I search for DatabaseHandle, I'd like to see line results for both /path/to/DatabaseHandle.js and code references like `import DatabaseHandle`. And I want to browse and search through them all at the same time, instead of running a code search command separate from a file search command.

I have that working. But now I'd like to show file previews, and I'm getting stuck on the behavior.
My problem arises because the search output differs between the two types of search commands which I'm using together.
Basically, the filesearch command returns a single token, that being the filepath.
The codesearch command returns multiple tokens - first is the filepath, second is the line number, third is the code line.

And I want to say, "if {2} exists, highlight the line using the value from {2}. If it's null, highlight line one (or don't highlight any line, which serves the same purpose)."

I'd really like to be able to either:
`--preview "bat --color=always --highlight-line ${{2}:-1} {1}`
or
`--preview " if [[ -z {2} ]]; then bat --color=always {1} else bat --color=always --highlight-line {2} {1}`
or something like that.

Nothing seemed like an obvious way to maybe-include a given token that might-not-exist. I've scoured the docs/issues/discussions and started trying to write a custom preview function to parse {}, but I think I'm getting tripped up parsing the quoted strings (and not having an easy time debugging when I need to fetch the tokens at runtime and somehow debug them by viewing variables in a preview window).

So I'd love any tips for how I should do this right!

**Comment by LangLangBart** [comment, marked as answer]
> I'd really like to be able to either:
> `--preview "bat --color=always --highlight-line ${{2}:-1} {1}`

Would this serve as a solution ?

```sh
cat <<'EOF' | fzf --delimiter ':' --preview 'line={2}; bat --color=always --highlight-line ${line:-1} {1}'
/usr/share/misc/ascii:8
/usr/share/misc/ascii
/usr/share/misc/birthtoken
/usr/share/misc/birthtoken:2
EOF
```

**Reply by danrlavoie** [reply]
Thank you, that works just as I intended! Looks like I just needed a temporary home for the token before writing out the `bat` command.
